libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -120,6 +122,70 @@ void AP_DDS_Client::update_topic(sensor_msgs_msg_NavSatFix& msg, const uint8_t i
     msg.position_covariance[8] = vdopSq;
 }
 
+void AP_DDS_Client::update_topic(sensor_msgs_msg_BatteryState& msg, const uint8_t instance)
+{
+    if (instance >= AP_BATT_MONITOR_MAX_INSTANCES) {
+        return;
+    }
+
+    update_topic(msg.header.stamp);
+    auto &battery = AP::battery();
```

Comment:
Before you get an instance of the battery, do you want to lock the HAL semaphore? 

Similar to the above `WITH_SEMAPHORE(gps.get_semaphore());`

This ensures the query is done all at the same timestamp, from what I understand.